ThermofluidStream/FlowControl/BasicControlValve.mo

```diff
@@ -31,6 +31,18 @@ protected
     else m_flow_ref_set/rho_ref "Reference volume flow";
 
 equation
+
+  //this if clause shall ensure that valid parameters have been entered
+  if flowCoefficient == FlowCoeffType.Kvs then
```

Comment:
@dzimmer @casella Considering only parameters are involved here, wouldn't it be better to place this `if` statement in an `initial equation` section, so it is only evaluated once instead of continuously? 
Or can it be expected (is it specified, maybe?) that tools optimize that out and only execute it once, anyway?

Reply:
I think that Dymola will optimize for it but the specification does not say anything about it.
Moving to initial equations seems like a good idea.

Reply:
Actually, the assertion level is set to "error". So the current model's evaluation shall be aborted. The simulation itself may continue with another evaluation, which IMO is not the case here.

Reply:
Stupid me... Of course that is why I did not care when implementing it.

Reply:
Huh? It will only abort if the assert fails, no? If not, does this not check if Kvs>0 over and over again?

Reply:
Meaning, if you have FlowCoeffType.Kvs, and a Kvs of, say, 0.1, you would still get an assert check at every time step?

Reply:
At least there is a Modelica example with assert() for parameters. 
https://build.openmodelica.org/Documentation/ModelicaReference.Operators.%27assert()%27.html

and assert for parameters is used e.g. in:

- Modelica.Blocks.Continuous.Filter
- Modelica.Electrical.Analog.Semiconductors.Diode2
- Modelica.Fluid.Pipes.BaseClasses.PartialStraightPipe
- Modelica.Mechanics.MultiBody.Joints.Revolute

I can't tell how often the assert is evaluated tho.

Reply:
- corrected some typos
- changed `V_flow_ref ` to `final parameter V_flow_ref`
- added a default value + assert for `d_valve` (`ThermofluidStream.FlowControl.SpecificValveType`). 

I didnt know how to suggest the changes, i hope its fine, that i directly pushed them.

 
Could we also change `m_flow_ref ` in `ThermofluidStream.FlowControl.Internal.PartialValve` to `parameter m_flow_ref`, or does that cause issues concerning conversion script or so?

Reply:
Pushing directly is the best way. 
I'll move the assert to the initial and if I do not find any reason against it would change m_flow_ref to a parameter